File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution(object):
    def frequencySort(self, s):
        """
        :type s: str
        :rtype: str
        """
        b=list(set(s))
        v=[]
        for i in b:
            v.append((s.count(i),i))
        v.sort(reverse=True)
        res=''
        for i in v:
            sam=i[1]*i[0]
            res=res+sam
            sam=''
        return res

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        
        unordered_map<char,int> mpp;

        for(int i=0;i<s.size();i++)
        {
            mpp[s[i]]++;
        }

        sort(s.begin(),s.end(),[&](char a, char b)
        {
            if(mpp[a]!=mpp[b])
                return mpp[a]>mpp[b];
            return a<b;
        });

        return s;
    }
};